packages/n/nana/xmake.lua

```diff
@@ -8,32 +8,61 @@ package("nana")
     add_versions("1.6.2", "5f5cb791dff292e27bfa29d850b93f809a0d91d6044ea7e22ce7ae76a5d8b24e")
     add_versions("1.7.2", "e2efb3b7619e4ef3b6de93f8afc70ff477ec6cabf4f9740f0d786904c790613f")
     add_versions("1.7.4", "56f7b1ed006c750fccf8ef15ab1e83f96751f2dfdcb68d93e5f712a6c9b58bcb")
+    if is_plat("linux") then
+        add_patches("1.7.4", path.join(os.scriptdir(), "patches", "1.7.4", "u8string_fix.patch"), "c783588816664124ba3b4077e18696899c8389419a015773b5bfe988e3a73f6a")
+    end
+
+    add_configs("nana_filesystem_force", {description = "Force nana filesystem over ISO and boost?", default = is_plat("linux"), type = "boolean"})
 
     if is_plat("linux", "windows") then
-        add_deps("cmake >=3.12")
+        add_deps("cmake >=3.15")
     end
 
     if is_plat("windows") then
         add_syslinks("ole32", "shell32", "kernel32", "user32", "gdi32", "winspool", "comdlg32", "advapi32")
         add_defines("_SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING")
     elseif is_plat("linux") then
-        add_syslinks("pthread", "X11", "Xft", "fontconfig")
+        add_deps("libxcursor")
+        add_syslinks("pthread", "fontconfig", "Xft")
     end
+    add_links("nana")
+
+    on_load("linux", "windows", function (package)
+        if package:config("nana_filesystem_force") then
+            package:add("defines", "NANA_FILESYSTEM_FORCE")
+        end
+    end)
 
     on_install("linux", "windows", function (package)
-        if is_plat("windows") then
+        if package:is_plat("windows") then
             local file_name = path.join(os.curdir(), "source", "system", "split_string.cpp")
             io.gsub(file_name, " and ", " && ")
         end
 
-        local configs = {"-DNANA_CMAKE_ENABLE_JPEG=OFF", "-DNANA_CMAKE_ENABLE_PNG=OFF"}
-        if is_plat("linux") then
+        local configs = {"-DNANA_CMAKE_ENABLE_JPEG=OFF", "-DNANA_CMAKE_ENABLE_PNG=OFF", "-DBUILD_SHARED_LIBS=OFF"}
+        if package:config("nana_filesystem_force") then
             table.insert(configs, "-DNANA_CMAKE_NANA_FILESYSTEM_FORCE=ON")
         end
-        import("package.tools.cmake").install(package, configs)
+        if package:is_plat("windows") then
+            local cmake_vsr = "MultiThreaded"
+            local pvsr = package:config("vs_runtime")
+            local d = package:config("debug")
+            if pvsr == "MT" and d then
+                cmake_vsr = "MultiThreadedDebug"
+            elseif pvsr == "MD" and not d then
+                cmake_vsr = "MultiThreadedDLL"
+            elseif pvsr == "MD" and d then
+                cmake_vsr = "MultiThreadedDebugDLL"
+            end
+            table.insert(configs, "-DMSVC_RUNTIME_LIBRARY=" .. cmake_vsr)
```

Comment:
we need not add them, xmake will pass vs runtime 

https://github.com/xmake-io/xmake/blob/8a1528fdb97a228ae62113d3b6c1b4e0a3e19381/xmake/modules/package/tools/cmake.lua#L231-L246

Reply:
I know. I found the code for it, but why are we getting the errors?

Reply:
I think I know why the CMAKE_MSVC_RUNTIME_LIBRARY macro doesn't work. On [this page](https://cmake.org/cmake/help/latest/policy/CMP0091.html#policy:CMP0091) at the bottom it states, that if the policy is not explicitly set it uses the OLD behaviour, which is ignoring the macro.

Reply:
I now added a patch which sets the policy and now all windows builds work. Maybe there is a way to fix this globally

Reply:
Global settings, there will be compatibility issues, not all libraries can support this new policy, there may be some conflicts.

We can only turn on this policy for specific packages.